base/server/src/main/java/com/netscape/cmscore/authentication/ChallengePhraseAuthentication.java

```diff
@@ -378,7 +369,7 @@ private String hashPassword(String pwd) {
         String salt = "lala123";
         byte[] pwdDigest = mSHADigest.digest((salt + pwd).getBytes());
         String b64E = Utils.base64encode(pwdDigest, true);
-
+        logger.debug("Password hashed with {}", mSHADigest.getAlgorithm());
         return "{SHA}" + b64E;
```

Comment:
I think you need to change "{SHA}" to a string that reflects the hash alg.  For example, something like "{SHA-256}".  I suspect you want to define these strings in CryptoUtil to retrieve based on the CryptoUtil.getDefaultHashAlgName().  I'm not sure if there is some kind of standard on the spelling though.  If it's up to us then could just use CryptoUtil.getDefaultHashAlgName() directly.  I'll let you find out.  But for backward compatibility (in case it matters), you'd at least keep the "{SHA}" spelling if it is sha1

Reply:
@ladycfu  thanks! I changed `{SHA}` to `mSHADigest.getAlgorithm().toUpperCase()` instead. This way if the algorithm does become user-defined in the future no changes are required here. I forced upper case as you suggested.